k8s/client.go

```diff
@@ -45,8 +65,15 @@ func New(apiServerURL string, caCert []byte, token string) (Client, error) {
 		return nil, fmt.Errorf("unable to parse ca certificate")
 	}
 
+	// same as net.DefaultTransport, with k8s CAs added
 	tr := &http.Transport{
 		TLSClientConfig: &tls.Config{RootCAs: pool},
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
```

Comment:
This is a large timeout. Is this because we long poll and thus receive no data on the socket?